File: backend/src/shared_lib/entities/best_practice.py
from pydantic import BaseModel

BestPracticeID = str


class BestPractice(BaseModel):
    id: BestPracticeID
    primary_id: str = ""
    label: str
    description: str
    risk: str
    status: bool
    results: list[str]
    hidden_results: list[str]


class BestPracticeInfo(BaseModel):
    id: int
    pillar: str
    question: str
    best_practice: dict[str, str]


class BestPracticeDto(BaseModel):
    status: bool | None = None


class RawBestPractice(BaseModel):
    primary_id: str
    label: str
    description: str
    risk: str

File: libs/backend/models/src/lib/BestPractice/BestPractice.ts
import type { SeverityType } from '../Finding';

export interface BestPractice {
  description: string;
  id: string;
  label: string;
  primaryId: string;
  results: Set<string>;
  risk: SeverityType;
  checked: boolean;
}

export interface BestPracticeBody {
  checked?: boolean;
}
